DQMOffline/Trigger/plugins/MuonMonitor.cc

```diff
@@ -0,0 +1,395 @@
+  setTitle(muonEtaME_variableBinning_,"Muon #eta","events");
+
+  histname = "muon_dxy"; histtitle = "Muon dxy";
+  bookME(ibooker,muondxy_,histname,histtitle, dxy_binning_.nbins, dxy_binning_.xmin,dxy_binning_.xmax);
+  setTitle(muondxy_,"Muon #dxy","events");
+
+  histname = "muon_dz"; histtitle = "Muon dz";
+  bookME(ibooker,muondz_,histname,histtitle, dxy_binning_.nbins, dxy_binning_.xmin,dxy_binning_.xmax);
+  setTitle(muondz_,"Muon #dz","events");
+
+
+  histname = "muon_etaphi"; histtitle = "Muon eta-phi"; 
+  bookME(ibooker,muonEtaPhiME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin, eta_binning_.xmax,phi_binning_1.nbins, phi_binning_1.xmin, phi_binning_1.xmax);
+  setTitle(muonEtaPhiME_,"#eta","#phi"); 
+
+  histname = "electron_pt_variable"; histtitle = "electron PT";
+  bookME(ibooker,eleME_variableBinning_,histname,histtitle,muon_variable_binning_);
+  setTitle(eleME_variableBinning_,"Electron pT [GeV]","events / [GeV]");
+
+  histname = "electron_eta"; histtitle = "electron eta";
+  bookME(ibooker,eleEtaME_,histname,histtitle, eta_binning_.nbins, eta_binning_.xmin,eta_binning_.xmax);
+  setTitle(eleEtaME_,"Electron #eta","events");
+
+  histname = "elePt_muPt"; histtitle = "electron pt vs muon pt";
+  bookME(ibooker,elePt_muPt_,histname,histtitle, elePt_variable_binning_2D_, muPt_variable_binning_2D_);
+  setTitle(elePt_muPt_,"electron pt [GeV]","muon pt [GeV]");
+
+  histname = "eleEta_muEta"; histtitle = "electron #eta vs muon #eta";
+  bookME(ibooker,eleEta_muEta_,histname,histtitle, eleEta_variable_binning_2D_, muEta_variable_binning_2D_);
+  setTitle(eleEta_muEta_,"electron #eta","muon #eta");
+
+
+
+  
+
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+void MuonMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+
+  edm::Handle<reco::PFMETCollection> metHandle;
+  iEvent.getByToken( metToken_, metHandle );
+  reco::PFMET pfmet = metHandle->front();
+  if ( ! metSelection_( pfmet ) ) return;
+  
+  //float met = pfmet.pt();
+  //  float phi = pfmet.phi();
+
+  edm::Handle<reco::VertexCollection> vtxHandle;
+  iEvent.getByToken(vtxToken_, vtxHandle);
+
+  reco::Vertex vtx;
+  math::XYZPoint pv(0, 0, 0);
+  for (vector<reco::Vertex>::const_iterator v = vtxHandle->begin(); v != vtxHandle->end(); ++v) {
```

Comment:
hi @gomber - two changes for this loop - first make it a range based loop (for const auto & v : *vtxHandle)

second - the copy to pv or vtx is not needed - just make vtx a pointer and use its position() method below where use pv.